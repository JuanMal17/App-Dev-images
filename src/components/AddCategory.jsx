import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputvalue ] = useState(['']);

    const onInputChange = ( {target}) => {
        setInputvalue( target.value )
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        // setCategories( categories => [ ...categories, inputValue]);
        onNewCategory( inputValue.trim() );
        setInputvalue('');
    }

    return (
        <form onSubmit={ onSubmit }>
            <input type='text' placeholder="Item" value={ inputValue }
            onChange={ onInputChange }/>
        </form>
    )
}