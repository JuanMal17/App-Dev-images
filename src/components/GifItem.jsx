import PropTypes from 'prop-types'

export const GifItem = ({ tittle, url }) => {

  return (
      <div className="card">
        <img src={ url } alt={ tittle } />
        <p>{ tittle }</p>
      </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url : PropTypes.string.isRequired,
}