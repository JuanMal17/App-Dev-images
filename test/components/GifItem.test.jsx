import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas GitItem ', () => {

  test('Match Snapshot ', () => {
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';
    const { container } = render(<GifItem title={title} url={url}/>);
    expect( container).toMatchSnapshot();
  });
  
});
