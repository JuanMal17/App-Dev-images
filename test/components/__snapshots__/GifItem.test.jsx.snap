// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GitItem  Match Snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://one-punch.com/saitama.jpg"
    />
    <p />
  </div>
</div>
`;
